app/models/api/trainee_filter_params.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Api
+  class TraineeFilterParams
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    ATTRIBUTES = %i[status since academic_cycle page per_page sort_order].freeze
+    ATTRIBUTES.each { |attr| attribute attr }
+
+    validate :check_statuses
+    validate :check_since
+    validate :check_academic_cycle
+    validates :sort_order, inclusion: %w[desc asc], allow_blank: true
+
+  private
+
+    def check_statuses
+      return if status.blank?
+
+      (status.is_a?(Array) ? status : [status]).each do |status_value|
+        errors.add(:status, "#{status_value} is not a valid status") unless Trainee.states.include?(status_value)
```

Comment:
Do we need to be checking against `TraineeFilter::STATUSES` here rather than `Trainee.states`?